### ISSUE TITLE
Reproduction Issue on MVTec

### ISSUE THREAD

**Issue by fl0308** [SkyShunsuke/InversionAD#7]
Hello,

Thank you for sharing this excellent work and the code.

I have successfully reproduced InversionAD on the MVTec AD dataset, but my results are lower than those reported in the paper. I would like to ask whether my results are within the expected range, or if there might be something wrong with my configuration.

My evaluation results are:

**Image-level Metrics**

* Average Image AUROC: 0.9539
* Average Image AP: 0.9797
* Average Image F1Max: 0.9471

**Pixel-level Metrics**

* Average Pixel AUROC: 0.9525
* Average Pixel AP: 0.3943
* Average PRO: 0.8682
* Average Pixel F1Max: 0.4625

**Efficiency Metrics**

* Average Latency: 979.81 ms
* Average Memory Usage: 13.59 GB

The configuration I used is:

```yaml
diffusion:
  model_type: 'dit'
  num_classes: 15
  target_channels: 3
  z_channels: 768
  depth: 16
  width: 2048
  num_sampling_steps: 100
  ema_decay: 0.999
  learn_sigma: false
  grad_checkpoint: false
  patch_size: 1
  channel_mult: [1,1,2,2]

backbone:
  model_type: efficientnet-b4
  outblocks: [1,5,9,21]
  outstrides: [2,4,8,16]
  pretrained: true
  stride: 16

data:
  batch_size: 8
  category: all
  data_root: /public/home/fl0308/feng/dataset/mvtec
  dataset_name: mvtec_ad_all
  img_size: 256
  multi_category: false
  num_normal_samples: -1
  num_workers: 1
  pin_memory: true
  train: true
  transform_type: imagenet

optimizer:
  grad_clip: 1.0
  init_lr: 1e-6
  peak_lr: 5e-5
  final_lr: 5e-6
  num_epochs: 300
  optimizer_name: adamw
  scheduler_type: warmup_cosine
  warmup_epochs: 40
  weight_decay: 0

evaluation:
  eval_interval: 25
  eval_step: 3

meta:
  seed: 42
  device: cuda
```

Could you please let me know whether these results are reasonable? If not, are there any important settings, checkpoints, or evaluation details that I might have missed?

Thank you very much for your help.

**Comment by SkyShunsuke** [comment]
Hi, 

Thank you for your attention to our work!
I think your experimental results may slightly deviate from our expected reproduction results. One possible reason is the number of GPUs you used. Since we used distributed training across 8 GPUs, it will affect the optimal learning rate and batch size. If you use a single GPU and have time to sweep these parameters, such hyperparameters will increase performance.

**Comment by fl0308** [comment]
Hi,

If I use a single NVIDIA A800 GPU with 32 CPU cores and 100 GB RAM, how would you recommend adjusting the training configuration to achieve the best performance?

In particular:

Which parameters should be modified compared to the default configuration？

**Comment by SkyShunsuke** [comment]
Hi, 
Thanks for sharing about your resources! I think you can first try to change the batch size (default 8) to 64. Also, monitoring performance at intermediate epochs might help to see if there are any loss spikes or degradation during pre-training.

**Comment by fl0308** [comment]
Thank you very much for your patient and detailed explanations. I will try the suggestions you provided.
I really appreciate your help and support.

I wish you all the best with your research and future work as well!

**Comment by SkyShunsuke** [comment]
Its my pleasure! If you have any updates on this issue, please feel free to let me know.
